### PR TITLE
nixos: append ccid as a plugin

### PR DESCRIPTION
**Pull request by domenkozar**
I got bitten by the fact that additional plugins removed the default one.

### CODE REVIEW

**Review by flokli — APPROVED** [review]
So I guess if I explicitly *not* want ccid support, I can still mkForce?

**Comment by domenkozar** [comment]
> So I guess if I explicitly _not_ want ccid support, I can still mkForce?

Yup!

File: nixos/modules/services/hardware/pcscd.nix [hunk 1]
```diff
@@ -56,6 +55,8 @@ in
     environment.systemPackages = [ package ];
     systemd.packages = [ (getBin package) ];
 
+    services.pcscd.plugins = [ pkgs.ccid ];
```
Comment (RaitoBezarius) [inline nixos/modules/services/hardware/pcscd.nix:58]: what is the rationale for not `lib.mkDefault`? wouldn't list merge?

Comment (domenkozar) [reply nixos/modules/services/hardware/pcscd.nix:58]: Lists are merged only at the same priority list.

Comment (RaitoBezarius) [reply nixos/modules/services/hardware/pcscd.nix:58]: Ah I understand better your PR message now :)

File: nixos/modules/services/hardware/pcscd.nix [hunk 2]
```diff
@@ -24,7 +24,6 @@ in
 
     plugins = mkOption {
       type = types.listOf types.package;
-      default = [ pkgs.ccid ];
       defaultText = literalExpression "[ pkgs.ccid ]";
```
Comment (RaitoBezarius) [inline nixos/modules/services/hardware/pcscd.nix:27]: You probably want to remove this I'd say because it's not technically a default per se, is it?

Comment (RaitoBezarius) [reply nixos/modules/services/hardware/pcscd.nix:27]: I guess that's fair but I don't see this pattern used :p cc @ncfavier who does a lot of work for nixos-search

Comment (ncfavier) [reply nixos/modules/services/hardware/pcscd.nix:27]: I think that's fine, in fact this pattern is why I made https://github.com/NixOS/nixpkgs/pull/230353. Not ideal obviously but I don't think we have a better way to indicate "the default value survives appending to the list with default override priority".

**Review by RaitoBezarius — APPROVED** [review]
LGTM :)

**Pull request merged**